aopy/visualization.py

```diff
@@ -261,18 +278,24 @@ def plot_spatial_map(data_map, x, y, alpha_map=None, ax=None, cmap='bwr'):
     cmap = copy.copy(matplotlib.cm.get_cmap(cmap))
     cmap.set_bad(color='black')
     
-    # Make an alpha map scaled between 0 and 1
-    if alpha_map is None:
-        alpha_map = 1
-    else:
+    # If an alpha map is present, make an rgba image
+    if alpha_map is not None:
+        if clim is None:
+            clim = (np.nanmin(data_map), np.nanmax(data_map))
+        norm = cm.colors.Normalize(*clim)
+        scalarMap = cm.ScalarMappable(norm=norm, cmap=cmap)
+        data_map = scalarMap.to_rgba(data_map)
+
+        # Apply the alpha map after scaling from 0 to 1
         alpha_range = np.nanmax(alpha_map) - np.nanmin(alpha_map)
         alpha_map = (alpha_map - np.nanmin(alpha_map)) / alpha_range
         alpha_map[np.isnan(alpha_map)] = 0
-
+        data_map[:,:,3] = alpha_map
```

Comment:
is it always 3? would it be more robust to add a new axis or something?

Reply:
yeah it's always 3, rgb are 0-2 and 3 is alpha. i think this is ok